library/ecp_curves.c

```diff
@@ -5417,27 +5420,57 @@ int mbedtls_ecp_mod_p521_raw(mbedtls_mpi_uint *X, size_t X_limbs)
  */
 static int ecp_mod_p255(mbedtls_mpi *N)
 {
-    mbedtls_mpi_uint Mp[P255_WIDTH];
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    size_t expected_width = 2 * ((255 + biL - 1) / biL);
```

Comment:
We are passing `expected_width` as  `X_Limbs` to `mbedtls_ecp_mod_p255_raw()`, where we are erroring out if `X_Limbs != 2 * P255_WIDTH`. It would be more robust to define `expected_width` in a way that is obviously correct:
```suggestion
    size_t expected_width = 2 * P255_WIDTH);
```

Reply:
Good catch, thanks.